src/registry.rs

```diff
@@ -38,13 +50,28 @@ fn cargo_home() -> Result<PathBuf> {
     Ok(cargo_home)
 }
 
-pub fn registry_url(manifest_path: &Path) -> Result<Url> {
+/// Find the URL of a registry
+pub fn registry_url(manifest_path: &Path, registry: Option<&str>) -> Result<Url> {
+    if env::var("CARGO_IS_TEST").is_ok() {
```

Comment:
why do we need this?

Reply:
I needed it to trick the `cargo add` tests into looking at crates-io, but it'll be unnecessary after I've added a non-default cargo config.